### PR TITLE
Fix test cases

### PR DESCRIPTION
**Pull request by yaohaizh**

### CODE REVIEW

File: azure-mgmt-resources/src/main/java/com/azure/management/RestClient.java [hunk 1]
```diff
@@ -8,31 +11,40 @@
...
+    private final URL baseUrl;
+
+    /**
+     * The original builder for this rest client.
+     */
+    private final RestClientBuilder builder;
 
-    public RestClient(RestAsyncClient client) {
-        this.client = client;
+    public RestClient(URL baseUrl, HttpPipeline pipeline, RestClientBuilder builder) {
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/main/java/com/azure/management/RestClient.java]: Not usual to take a Builder as parameter for constructor. And if we go from Builder, we probably do not need a public constructor for RestClient.

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestUtilities.java [hunk 2]
```diff
@@ -24,4 +27,12 @@ public static void sleep(int milliseconds, boolean isRecordMode) {
         }
     }
 
+    public static <T> int getPagedIterableSie(PagedIterable<T> iterable) {
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestUtilities.java]: Size ?

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/fluentcore/dag/ProxyTaskGroupTests.java [hunk 3]
```diff
@@ -1451,10 +1452,14 @@ public void beforeGroupInvoke() {
...
         final ArrayList<String> seen = new ArrayList<>();
+        CountDownLatch down = new CountDownLatch(1);
         itiC.getTaskGroup()
                 .invokeAsync(itiC.getTaskGroup().newInvocationContext())
                 .subscribe(indexable ->
-                        seen.add(indexable.getKey()));
+                                seen.add(indexable.getKey()),
+                        throwable -> down.countDown(),
+                        () -> down.countDown());
+        down.await();
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/fluentcore/dag/ProxyTaskGroupTests.java]: Should we use await with timeout?

Comment (yaohaizh) [reply azure-mgmt-resources/src/test/java/com/azure/management/resources/fluentcore/dag/ProxyTaskGroupTests.java]: Without network connection, just keep the case simple.

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestUtilities.java (hunk 2 above)
Comment (yaohaizh) [reply azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestUtilities.java]: Fixed

File: azure-mgmt-resources/src/main/java/com/azure/management/RestClient.java (hunk 1 above)
Comment (yaohaizh) [reply azure-mgmt-resources/src/main/java/com/azure/management/RestClient.java]: Agree, this constructor is migrated all the V1.  We need sort out of the build/serviceclient relationship.

**Pull request merged**